### ISSUE TITLE
Host Hashcat parameters

### ISSUE THREAD

**Issue by soxrok2212** [nesfit/fitcrack#27]
An additional question/feature request (as I have not seen it yet) - is it possible to add extra parameters to hosts, eg for a dedicated server, `-w 4` or for a home user's machine, maybe `-w 2` and things like that?

**Comment by ihranicky** [comment]
Hi, this feature is already supported, but was not described in the main readme/manual. I added the description here (dev branch): https://github.com/nesfit/fitcrack/blob/dev/README.md

Simply create a file Simply create a file `/etc/fitcrack.conf` (Linux host) or `C:\ProgramData\BOINC\fitcrack.conf` (Windows host) and put the `-w 4` inside :)

**Comment by soxrok2212** [comment]
Thanks for the tip! In hashtopolis, this can be configured from the administration page. Is that a feasible feature to be added?

**Comment by davidbolvansky** [comment]
Yeah, we should add proper UI settings :)

**Comment by neuton** [comment]
Should be possible to add hashcat parameters via server attack config. For example, --hex-wordlist seems to be the only possible way to brute force a custom mask with fixed non-regular characters (arbitrary bytes rather than printable ascii)

**Comment by davidbolvansky** [comment]
> Should be possible to add hashcat parameters via server attack config. For example, --hex-wordlist seems to be the only possible way to brute force a custom mask with fixed non-regular characters (arbitrary bytes rather than printable ascii)

--hex-wordlist applies to dictionaries, no? Nothing with mask attack, or?

**Comment by neuton** [comment]
> > Should be possible to add hashcat parameters via server attack config. For example, --hex-wordlist seems to be the only possible way to brute force a custom mask with fixed non-regular characters (arbitrary bytes rather than printable ascii)
> 
> --hex-wordlist applies to dictionaries, no? Nothing with mask attack, or? Maybe you mean hybrid attacks?
> 
> Can you please post more info (settings, configuration) about the job which fails for you?

My task is to brute-force a raw byte (non-ascii) string with part of the known bytes fixed in place. And I simply don't know any other way to run this, apart from using the hybrid attack with the dictionary and the --hex-wordlist argument. I may be mistaken though.

**Comment by davidbolvansky** [comment]
Right, so hybrid attack.

Should not be hard to implement, I will consider it.

**Comment by davidbolvansky** [comment]
Try dev branch.

When uploading your dictionary, mark it as HEX dict.

![image](https://user-images.githubusercontent.com/22410393/186208453-ec0d80f1-8018-4d60-ac95-6a1f7c7f124d.png)

Then in hybrid or dict attack modes --hex-wordlist should be added automatically.

**Comment by davidbolvansky** [comment]
> Yeah, we should add proper UI settings :)

In plans for next release, most likely 2.4.1

**Comment by davidbolvansky** [comment]
@soxrok2212 I think workload profile and device types could be there, in host settings. Or any other ideas, tips?

**Comment by davidbolvansky** [comment]
Device types added in dev branch.

![image](https://github.com/nesfit/fitcrack/assets/22410393/4e9ae1de-f4fc-44dc-a256-8c10156e46db)

**Comment by davidbolvansky** [comment]
> An additional question/feature request (as I have not seen it yet) - is it possible to add extra parameters to hosts, eg for a dedicated server, -w 4 or for a home user's machine, maybe -w 2 and things like that?

Yes, second part of this feature should be done on host settings page.

**Comment by davidbolvansky** [comment]
Workload profile now available from Add Job page.

![image](https://github.com/nesfit/fitcrack/assets/22410393/a52f1e7b-faf8-452e-b3dd-f9ad4ee872e7)


Host settings to be added soon.
